ruma-common/src/push.rs

```diff
@@ -14,19 +14,18 @@
 //! - sender rules
 //! - underride rules
 //!
-//! Each of these kind of rule has a corresponding type that is
-//! just a wrapper around another type:
+//! Each of these kind of rule has a corresponding `AnyPushRule` variant that is
+//! just a wrapper around a type:
 //!
 //! - `SimplePushRule` for room and sender rules
 //! - `ConditionalPushRule` for override and underride rules: push rules that may depend on a
 //!   condition
 //! - `PatternedPushRules` for content rules, that can filter events based on a pattern to trigger
 //!   the rule or not
 //!
-//! Having these wrapper types allows to tell at the type level what kind of rule you are
-//! handling, and makes sure the `Ruleset::add` method adds your rule to the correct field
-//! of `Ruleset`, and that rules that are not of the same kind are never mixed even if they share
-//! the same representation.
+//! Having these wrapper variants allows to tell what kind of rule you are handling, and makes sure
+//! the `Ruleset::add` method adds your rule to the correct field of `Ruleset`, and that rules that
+//! are not of the same kind are never mixed even if they share the same representation.
 //!
 //! It is still possible to write code that is generic over a representation by manipulating
 //! `SimplePushRule`, `ConditonalPushRule` or `PatternedPushRule` directly, instead of the wrappers.
```

Comment:
I think with the field type changes reverted, all of this just adds confusion and should be removed.